bindings/cs/MongoDB.Crypt/CryptOptions.cs

```diff
@@ -14,28 +14,55 @@
  * limitations under the License.
  */
 
+using System.Collections.Generic;
+using System.Linq;
 
 namespace MongoDB.Crypt
 {
     /// <summary>
-    /// Options to configure mongocrypt with
+    /// Options to configure mongocrypt with.
     /// </summary>
     public class CryptOptions
     {
-        public IKmsCredentials KmsCredentials { get; }
+        public Dictionary<KmsType, IKmsCredentials> KmsCredentialsOptions { get; } = new Dictionary<KmsType, IKmsCredentials>();
         public byte[] Schema { get; }
 
-        public CryptOptions(IKmsCredentials kmsCredentials)
+        public CryptOptions(IKmsCredentials kmsCredentials) : this(kmsCredentials, null)
         {
-            KmsCredentials = kmsCredentials;
         }
 
-        public CryptOptions(IKmsCredentials kmsCredentials, byte[] schema)
+        public CryptOptions(IKmsCredentials kmsCredential, byte[] schema) : this(new[] { kmsCredential }, schema)
         {
-            KmsCredentials = kmsCredentials;
+        }
+
+        public CryptOptions(IEnumerable<IKmsCredentials> kmsCredentialsOptions) : this(kmsCredentialsOptions, null)
+        {
+        }
+
+        public CryptOptions(IEnumerable<IKmsCredentials> kmsCredentialsOptions, byte[] schema)
+        {
+            InitializeKmsCredentialsOptions(kmsCredentialsOptions.ToList());
             Schema = schema;
         }
 
+        private void InitializeKmsCredentialsOptions(List<IKmsCredentials> kmsCredentialsOptions)
+        {
+            if (kmsCredentialsOptions != null && kmsCredentialsOptions.Count != 0)
+            {
+                foreach (var kmsCredential in kmsCredentialsOptions)
+                {
+                    if (KmsCredentialsOptions.ContainsKey(kmsCredential.KmsType))
+                    {
+                        KmsCredentialsOptions[kmsCredential.KmsType] = kmsCredential;
```

Comment:
Is it possible to have more than one credential of a particular type? @jyemin

Reply:
No.

Reply:
If it's illegal to have more than one credential of a particular type, should we throw an exception in the case that this happens? Or is it specified somewhere that we should just quietly replace the old value with the new value? @jyemin @DmitryLukyanov

Reply:
As I see in `java` implementation, we just overwrite the previous value: https://github.com/mongodb/libmongocrypt/blob/master/bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptOptions.java#L88 via a builder logic

Reply:
So, maybe it's enough to have the current version?

Reply:
Right, Java does overwrite the previous value. They are using the builder pattern though, which makes it very explicit that you're "re-setting" an option if you specify more than one local or aws credential. Here, we're taking in a list of credentials, and it's non-obvious IMHO that only 1 of each type is allowed.  How do you feel about making this restriction more explicit? (see my comments for the constructors)

Reply:
Resolved per offline discussion